components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -553,61 +560,133 @@ private HashMap getAPIPermissionArray(String permissionJsonString) throws ParseE
      * @param apiId            UUID of the API.
      * @param status           Target life cycle status.
      * @param checkListItemMap Map of check list items.
+     * @return WorkflowResponse workflow response related to LC state change.
      * @throws APIManagementException If failed to update API lifecycle status..
      */
     @Override
```

Comment:
As this is a Impl class, shall we change the method comment as in - https://github.com/chamindias/carbon-apimgt/blob/b64c0402b3d5cd11b3f3841e4ea648a7a0254ec5/components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java#L243. Basic idea of this is to avoid duplicating the parent method comment. With @see annotation you can link to the parent class method.